sdk/src/main/java/com/qonversion/android/sdk/internal/QonversionInternal.kt

```diff
@@ -107,6 +110,23 @@ internal class QonversionInternal(
         })
     }
 
+    override fun syncHistoricalData() {
+        val isHistoricalDataSynced: Boolean = sharedPreferencesCache?.getBool(Constants.IS_HISTORICAL_DATA_SYNCED) ?: false
+        if (isHistoricalDataSynced) {
+            return
+        }
+
+        Qonversion.shared.restore(callback = object: QonversionEntitlementsCallback {
+            override fun onSuccess(entitlements: Map<String, QEntitlement>) {
+                sharedPreferencesCache?.putBool(Constants.IS_HISTORICAL_DATA_SYNCED, true)
```

Comment:
maybe worth logging something on success too?

Reply:
I don't think so. I don't see any reason to print something here.